src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
@wateret thank you. I think the esp at the push eax is actually not correct and it should be larger by 4. But we cannot do that, since we would overwrite something on the stack of the CallEHFunclet (or any other native method whose context is passed to ThrowExceptionFromContextInternal.
Can you please change the ThrowExceptionFromContextInternal calling convention to __fastcall and pass the PAL_SEHException * in ecx? I believe that should fix the issue.